src/Storages/StorageFile.cpp

```diff
@@ -115,8 +115,12 @@ void listFilesWithRegexpMatchingImpl(
     {
         try
         {
-            fs::path path = fs::canonical(path_for_ls + for_match);
-            result.push_back(path.string());
+            /// We use fs::canonical to check if the file exists but the result path
```

Comment:
That's my first solution, but then a test failed because fs::absolute only checks if the symlink exists, while originally fs::canonical checks both symlink and the linked file exist.

Reply:
Please add a comment and a test that checks that (that it fails gracefully I guess if the target of the symlink doesn't exist). Otherwise it will get broken again in the future

Reply:
> add a test that checks that (that it fails gracefully I guess if the target of the symlink doesn't exist)

It's already in `02771_multidirectory_globs_storage_file`